xds/internal/xdsclient/client.go

```diff
@@ -42,17 +43,37 @@ type clientImpl struct {
 	done   *grpcsync.Event
 	config *bootstrap.Config
 
-	controller controllerInterface
-
-	logger *grpclog.PrefixLogger
-	pubsub *pubsub.Pubsub
+	// authorityMu protects the authority fields. It necessary because an
+	// authority is created when it's used.
+	authorityMu sync.Mutex
+	// authorityPerConfig is a map from ServerConfig to authority. So that
+	// different authorities sharing the same ServerConfig can share the
+	// authority. The key is ServerConfig.String().
+	//
+	// An authority is either in authorityPerConfig, or idleAuthorityPerConfig,
+	// never both.
+	authorityPerConfig map[string]*authority
```

Comment:
Optional:

s/authorityPerConfig/authorities/ (or authoritiesMap)
s/idleAuthorityPerConfig/idleAuthorities/ (or idleAuthoritiesMap)

The plural in the name is more appropriate for a map, and it will be clear at call sites that we are indexing the map using the server config (so omitting that in the name should be fine I think).

Reply:
Done